core/modules/comment/comment.theme.inc

```diff
@@ -107,7 +107,7 @@ function template_preprocess_comment(&$variables) {
 
   $variables['title'] = l($comment->subject, $uri['path'], $uri['options']);
   $variables['permalink'] = l(t('Permalink'), $uri['path'], $uri['options']);
-  $variables['submitted'] = t('!datetime by !username', array('!datetime' => $variables['created'], '!username' => $variables['author']));
+  $variables['submitted'] = token_replace($node_type->settings['comment_submitted_format'], array('comment' => $comment));
```

Comment:
```suggestion
  $variables['submitted'] = filter_xss_admin(token_replace($node_type->settings['comment_submitted_format'], array('comment' => $comment)));
```

Reply:
Turns out this is not needed because `token_replace()` defaults to using `$options['sanitize'] = TRUE`.